client/components/data/query-user-devices/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestUserDevices } from 'state/user-devices/actions';
+import { isRequestingUserDevices } from 'state/selectors';
+
+class QueryUserDevices extends Component {
+	static propTypes = {
+		isRequesting: PropTypes.bool,
+		requestUserDevices: PropTypes.func
+	}
+
+	componentWillMount() {
+		if ( this.props.requesting ) {
```

Comment:
If we're using the `http` layer, I don't think it's necessary to check if we already have a request in-flight. We should already be filtering these gets, with the `remove-duplicate-gets` pipeline handler see: https://github.com/Automattic/wp-calypso/blob/master/client/state/data-layer/wpcom-http/pipeline/index.js#L53

cc @dmsnell